src/core/alert/AlertProvider.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { createContext, useContext, useState } from "react";
+
+import { Alert } from "@components"
+
+import { AlertsContainer } from "./style"
+
+type Alert = {
+    msg: string,
+    id: number,
+    variant: boolean
+}
+
+interface Props {
+    alerts: Alert[],
+    addAlert: (msg: string) => void
+}
+
+const STATE: Props = {
+    alerts: [],
+    addAlert: () => { }
+}
+
+const Context = createContext(STATE);
+
+export const AlertProvider = (props: any) => {
+    const [alerts, setAlerts] = useState<Alert[]>([]);
+
+    // const sleep = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));
+
+    const addAlert = (msg: string) => {
+        const id = alerts.length > 0 ? alerts[alerts.length - 1].id + 1 : 0;
+        setAlerts([...alerts, { msg: msg, id: id, variant: false }]);
+    }
+
+    const removeAlert = async (id: number) => {
+        // changeVisibility(id);
+        // await sleep(1000);
+        setAlerts(prev => prev.filter(e => e.id !== id))
+    }
+
+    // const changeVisibility = (id: number) => {
+    //     let items = alerts;
+    //     items.map(e => {
+    //         if (e.id === id) {
+    //             e.variant = !e.variant
+    //         }
+    //     })
+    //     setAlerts([...items]);
+    // }
+
+    const state: Props = {
+        alerts: alerts,
+        addAlert: addAlert
+    }
+
+    return (
+        <Context.Provider value={state}>
+            {alerts.length > 0 &&
+                <AlertsContainer>
+                    {alerts.map((e) => {
```

Comment:
change e to something meaningful (e indicates event, whereas it's alert obj) and add second arg index provided with .map function

Reply:
In map function e is not an event, this is element...